src/MongoDB.Bson/Serialization/Serializers/DiscriminatedWrapperSerializer.cs

```diff
@@ -76,7 +76,7 @@ public override TValue Deserialize(BsonDeserializationContext context, BsonDeser
             var bsonReader = context.Reader;
             var nominalType = args.NominalType;
             var actualType = _discriminatorConvention.GetActualType(bsonReader, nominalType);
-            var serializer = BsonSerializer.LookupSerializer(actualType);
+            var serializer = _wrappedSerializer;
```

Comment:
I'm suspicious of this change.

The whole point of this serializer is to use the discriminator to determine the actual type and then find the correct serializer for the ACTUAL type.

Not sure how using the _wrappedSerializer works. Maybe it only works for SOME wrapped serializers.